sdk/python/lib/pulumi/output.py

```diff
@@ -440,6 +440,23 @@ def contains_unknowns(val: Any) -> bool:
     return rpc.contains_unknowns(val)
 
 
+def _is_prompt(value: Input[T]) -> bool:
```

Comment:
Am I missing something similar here in the SDK? This was a bit non-trivial to write as I'm wondering if I understand the `is_input_type` clause and should I be checking anything else.

Reply:
I don't think we should be checking `is_input_type` here -- all that does is check to see if it's a class that has the `@input_type` decorator applied. It's unrelated to `Input[T]`.

Reply:
Agreed, after reflecting about it and writing my "haskeller's impression" doc, this check is not needed here. The entire thing makes a bit more sense now.